From b788299f1b0dd47c3b969dfeb2ff97e05a26fde4 Mon Sep 17 00:00:00 2001
From: Masahiro Yamada <yamada.masahiro@socionext.com>
Date: Wed, 12 Sep 2018 15:43:29 +0900
Subject: [PATCH 1/3] kbuild: hide most of targets when running config or mixed
 targets

commit 25815cf5ffecfb8b54314ee1b7cf14c78681fbb6 upstream.

When mixed/config targets are being processed, the top Makefile
does not need to parse the rest of targets.

Signed-off-by: Masahiro Yamada <yamada.masahiro@socionext.com>
Signed-off-by: Li Zhou <li.zhou@windriver.com>
---
 Makefile | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 7a4a96452208..4faf3752d2e5 100644
--- a/Makefile
+++ b/Makefile
@@ -1600,9 +1600,6 @@ namespacecheck:
 export_report:
 	$(PERL) $(srctree)/scripts/export_report.pl
 
-endif #ifeq ($(config-targets),1)
-endif #ifeq ($(mixed-targets),1)
-
 PHONY += checkstack kernelrelease kernelversion image_name
 
 # UML needs a little special treatment here.  It wants to use the host
@@ -1717,6 +1714,8 @@ ifneq ($(cmd_files),)
   include $(cmd_files)
 endif
 
+endif   # ifeq ($(config-targets),1)
+endif   # ifeq ($(mixed-targets),1)
 endif	# skip-makefile
 
 PHONY += FORCE
-- 
2.17.1

